### ISSUE TITLE
Data Table <TableSelectAll> will not render based on example code from carbon

### ISSUE THREAD

**Issue by davekaj** [carbon-design-system/carbon-components-react#724]
## Detailed description

The error I am having can be seen in the screenshot below. It appears that the issue ends up being in the Inlinecheckbox component. the code is also pasted below, I have commented out the functions that would not let me render because I have not created them yet, but otherwise they are the exact same as the example code 

> Describe in detail the issue you're having.

The sorting Data Table will not render

> Is this a feature request (new component, new icon), a bug, or a general issue?

Bug

> Is this issue related to a specific component?

Yes

> What did you expect to happen? What happened instead? What would you like to see changed?

I expected the data table component to render properly based on the example code from https://github.com/carbon-design-system/carbon-components-react/tree/master/src/components/DataTable 

> What browser are you working in?

chrome

> What version of the Carbon Design System are you using?

both 5.31.1 and 5.31.0 have been tried

## Steps to reproduce the issue

1. Download and set up carbon
2. Import data table component
3. Try to render the Selection use cases example of data table 
4. Get the error (see image below)

![image](https://user-images.githubusercontent.com/22417382/37609363-50e1ced0-2b73-11e8-88d5-3933772cd780.png)

          <DataTable
            rows={FakeTableData}
            headers={FakeTableHeaders}
            render={({
              rows,
              headers,
              getHeaderProps,
              getSelectionProps,
              getBatchActionProps,
              onInputChange,
              selectedRows,
            }) => (
              <TableContainer title='DataTable with batch actions'>
                <TableToolbar>
                  {/* make sure to apply getBatchActionProps so that the bar renders */}
                  <TableBatchActions {...getBatchActionProps()}>
                    {/* inside of you batch actinos, you can include selectedRows */}
                    <TableBatchAction >
                        Ghost
                    </TableBatchAction>
                    <TableBatchAction >
                        Ghost
                    </TableBatchAction>
                    <TableBatchAction >
                        Ghost
                    </TableBatchAction>
                  </TableBatchActions>
                  <TableToolbarSearch onChange={onInputChange} />
                  <TableToolbarContent>
                    <TableToolbarAction
                      iconName='download'
                      iconDescription='Download'
                      // onClick={action('TableToolbarAction - Download')}
                    />
                    <TableToolbarAction
                      iconName='edit'
                      iconDescription='Edit'
                      // onClick={action('TableToolbarAction - Edit')}
                    />
                    <TableToolbarAction
                      iconName='settings'
                      iconDescription='Settings'
                      // onClick={action('TableToolbarAction - Settings')}
                    />
                    <Button
                      // onClick={action('Add new row')} 
                      small
                      kind='primary'
                    >
                        Add new
                    </Button>
                  </TableToolbarContent>
                </TableToolbar>
                <Table>
                  <TableHead>
                    <TableRow>
                      <TableSelectAll {...getSelectionProps()} />
                      {headers.map((header) => (
                        <TableHeader {...getHeaderProps({ header })}>
                          {header.header}
                        </TableHeader>
                      ))}
                    </TableRow>
                  </TableHead>
                  <TableBody>
                    {rows.map((row) => (
                      <TableRow key={row.id}>
                        <TableSelectRow {...getSelectionProps({ row })} />
                        {row.cells.map((cell) => (
                          <TableCell key={cell.id}>{cell.value}</TableCell>
                        ))}
                      </TableRow>
                    ))}
                  </TableBody>
                </Table>
              </TableContainer>
            )}

**Comment by joshblack** [comment]
Hi @davekaj! 👋 Thanks for taking the time to make this issue. Am having a hard time reproducing as the storybook works as expected with that snippet above. Could you share a link to the file where you're trying to use this component so we can help debug further? Can share here, or through slack, depending on confidentiality 👍

**Comment by Adam0000** [comment]
It looks like the cause is InlineCheckbox using Fragment, which is a feature introduced in React 16.2.

https://github.com/carbon-design-system/carbon-components-react/blob/0e039c2c83231af107c83ef7affeec21b1d479ee/src/components/InlineCheckbox/InlineCheckbox.js#L12-L23

**Comment by davekaj** [comment]
yep , i was using react 15.6 . i believe this is the fix

**Comment by Adam0000** [comment]
@joshblack We were able to upgrade React to avoid this issue, but is it your intention to also support earlier versions of React? If so maybe we should reopen the issue.

**Comment by joshblack** [comment]
@Adam0000 for the jump from React v15 -> v16, I think we would prefer to help teams upgrade instead of providing backwards support. For certain components, I think backwards compatibility is essential though so we've exposed those where appropriate. With newer components, I think the tendency will be to support React v16.2 and v16.3 given all the new stuff that's coming in with these releases that will really help our process.

If you feel like this direction is misguided, please feel free to speak up though. If there is a large enough need to support older versions than I'm confident we'll find a way to make it happen 😄

**Comment by Adam0000** [comment]
@joshblack Given how easy it is to upgrade to React 16, your strategy is probably best.

**Comment by oliverodaa** [comment]
Commenting here for the sake of someone who might have the same bug as I just did.

I had the same error with `<InlineCheckbox>` so I updated the React version to `16.2.0`.

Then I started getting this error, and for the life of me could not identify what the issue was:

```
Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: symbol.
```

The trick is that I also had to update `react-dom` when you update `react`.

Hope that helps someone!
